### PR TITLE
docs: update glossary

### PR DESCRIPTION
**Pull request by erickzhao**
#### Description of Change

We have a handy glossary that we haven't updated in a while!

New entries: code signing, context isolation, MSI, notarization, preload script, sandbox
Removed entries: libchromiumcontent
Modified entries: ASAR, DMG, NSIS, process

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -85,22 +116,40 @@ likely to use a different V8 version from the Node binary installed in your
...
+See also: [code signing](#code-signing)
 
 ### NSIS
 
 Nullsoft Scriptable Install System is a script-driven Installer
 authoring tool for Microsoft Windows. It is released under a combination of
 free software licenses, and is a widely-used alternative to commercial
-proprietary products like InstallShield. [electron-builder] supports NSIS
-as a build target.
+proprietary products like InstallShield.
```
Comment (malept) [inline docs/glossary.md:134]: Arguably we can just delete this since it's not even referenced anywhere in our docs.

File: docs/glossary.md [hunk 2]
```diff
@@ -4,24 +4,47 @@ This page defines some terminology that is commonly used in Electron development
...
+
+### code signing
+
+Code signing is a process where an app developer digitally signs their code to
+ensure that it hasn't been tampered with after packaging. Both Windows and
+macOS implement their own version of code signing. As a desktop app developer,
+it's important that you sign your code if you plan on distributing it to the
+general public.
+
+For more information, read the [Code Signing] tutorial.
```
Comment (malept) [inline docs/glossary.md:23]: TIL at least GitHub links references case-insensitively.

Comment (erickzhao) [reply docs/glossary.md:23]: Yep. I also checked locally that it works with MDX on the new site as well!

**Review by malept — COMMENTED** [review]
The consistency on this document is practically nonexistent (although that's not your fault). Do we want to address that now or in a follow-up PR?

Otherwise this is fine.

**Comment by erickzhao** [comment]
> The consistency on this document is practically nonexistent (although that's not your fault). Do we want to address that now or in a follow-up PR?

What do you mean by consistency in particular here? I thought it was decently-formatted.

**Comment by malept** [comment]
How each definition is worded. Mostly, it's pretty obvious that multiple folks wrote the definitions.

**Review by zeke — APPROVED** [review]
👋🏼 👍🏼

**Review by codebytere — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Pull request merged**